requirements/requirements.txt

```diff
@@ -0,0 +1,77 @@
+#
+# This file is autogenerated by pip-compile with Python 3.9
+# by the following command:
+#
+#    pip-compile --output-file=requirements/requirements.txt pyproject.toml
+#
+attrs==23.1.0
+    # via jsonlines
+certifi==2023.5.7
+    # via requests
+charset-normalizer==3.1.0
+    # via requests
+filelock==3.12.2
+    # via
+    #   huggingface-hub
+    #   torch
+    #   transformers
+fsspec==2023.6.0
+    # via huggingface-hub
+h5py==3.8.0
+    # via sambanova-generative-data-prep (pyproject.toml)
+huggingface-hub==0.15.1
+    # via transformers
+idna==3.4
+    # via requests
+jinja2==3.1.2
+    # via torch
+jsonlines==3.1.0
+    # via sambanova-generative-data-prep (pyproject.toml)
+markupsafe==2.1.3
+    # via jinja2
+mpmath==1.3.0
+    # via sympy
+networkx==3.1
+    # via torch
+numpy==1.24.3
+    # via
+    #   h5py
+    #   sambanova-generative-data-prep (pyproject.toml)
+    #   transformers
+packaging==23.1
+    # via
+    #   huggingface-hub
+    #   transformers
+psutil==5.9.5
+    # via sambanova-generative-data-prep (pyproject.toml)
+pyyaml==6.0
+    # via
+    #   huggingface-hub
+    #   transformers
+regex==2023.6.3
+    # via transformers
+requests==2.31.0
+    # via
+    #   huggingface-hub
+    #   transformers
+safetensors==0.3.1
+    # via transformers
+sympy==1.12
+    # via torch
+tokenizers==0.13.3
+    # via transformers
+torch==2.0.1
+    # via sambanova-generative-data-prep (pyproject.toml)
+tqdm==4.65.0
+    # via
+    #   huggingface-hub
+    #   sambanova-generative-data-prep (pyproject.toml)
+    #   transformers
+transformers==4.30.2
+    # via sambanova-generative-data-prep (pyproject.toml)
+typing-extensions==4.6.3
+    # via
+    #   huggingface-hub
+    #   torch
+urllib3==2.0.3
+    # via requests
```

Comment:
Previously we did not require specific versions of requirements so that they did not need to have a specific version of python. I am worried that with such recent versions of all the packages they will require a very new version of python? Ideally we want to make it compatible with python 3.7 ->.

I do not know a good solution to this at all, suggestions would be super appreciated!

Reply:
These are for distribution, not for development. `pyproject.toml` have the requirements for developers, the files in `requirements` is to ensure our distribution works the same for everyone.